### ISSUE TITLE
no_weight_decay_on_bn removes weight decay on FC

### ISSUE THREAD

**Issue by nagisa-eevee** [hysts/pytorch_image_classification#31]
I notice that if `no_weight_decay_on_bn` is set to `True`, weight decay will only apply to `conv.weight`. It seems that weight decay on fc layers are also removed at the same time. Is there any reason to do so?

**Comment by hysts** [comment]
Hi, @nagisa-eevee 

I think I added the option to try what's written in section 4.2 of [this paper](https://arxiv.org/abs/1807.11205), and if I remember correctly, in my experiments, the accuracy was higher when no weight decay was applied to the fc layer. So that's the reason.
But the accuracy drop may be because I used the same parameter for weight decay on conv and fc layers, and tuning weight decay parameter on fc layer could lead to better accuracy. So in that sense, the current implementation that forces the weight decay on the fc layer to be 0 is not very good.
